### PR TITLE
Enable vagrant functional testing

### PR DESCRIPTION
**Pull request by chambridge**
Enable vagrant functional testing.
Remove files from git that should not be in repository.
Fix dependencies on checked in files.
Add files to .gitignore and fix run_cmds to handle case where commands don't provide results on the system.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -36,7 +18,45 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
\ No newline at end of file
+
+    # Define a test box to be scanned during tests
+    config.vm.define "test_2" do |test_2|
+          test_2.vm.network "private_network", ip: "192.168.50.11"
+          test_2.vm.host_name = "test2.example.com"
+
+          test_2.ssh.forward_x11 = true
+
+          test_2.vm.provider :libvirt do |domain|
```
Comment (noahl) [inline Vagrantfile]: Are we still using this block? I think both of us are using VirtualBox, so I'm not sure this code is ever called.

File: Vagrantfile [hunk 2]
```diff
@@ -36,7 +18,45 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
+      end
+
+      # Define a test box to be scanned during tests
+      config.vm.define "test_3" do |test_3|
+            test_3.vm.network "private_network", ip: "192.168.50.12"
+            test_3.vm.host_name = "test3.example.com"
+
+            test_3.ssh.forward_x11 = true
+
+            test_3.vm.provider :libvirt do |domain|
```
Comment (noahl) [inline Vagrantfile]: Same as previous comment.

File: rho/clicommands.py [hunk 3]
```diff
@@ -106,7 +106,12 @@ def _read_in_file(filename):
 
 # Write new credentials in the related file.
 def _save_cred(cred):
-    with open('data/credentials', 'a') as cred_file:
+    append_or_write = 'a'
+    if not os.path.exists('data'):
+        os.makedirs('data')
+    if not os.path.isfile('data/credentials'):
```
Comment (noahl) [inline rho/clicommands.py:112]: Suggestion:
```
if os.path.isfile('data/credentials'):
    append_or_write = 'a'
else:
    append_or_write = 'w'
```

**Review by noahl — APPROVED** [review]
I left some small suggestions, but this is a great change, and I'm looking forward to getting it in. I'm especially excited about removing all of these files from the repo.

File: Vagrantfile (hunk 1 above)
Comment (chambridge) [reply Vagrantfile]: I'll remove this. I didn't really educate myself on Vagrant. Just essentially kept, what was there for the test endpoints. But I've tested with out this portion and it works. I'll also remove the libvrt section in the doc.

File: Vagrantfile (hunk 2 above)
Comment (chambridge) [reply Vagrantfile]: removed.

**Pull request merged**